### PR TITLE
BREAKING CHANGE(web-twig): Rename `message` prop to `validationText` …

### PR DESCRIPTION
**Pull request by literat**
…in Form Fields #DS-676

And also make it depend on `validationState` and introduce ValidationText component.

Improve demos or related components.

Instead of `message` prop use `validationText`. Use `helperText` instead of `validationText` to show permanent helper. If `validationState` is not set `validationText` won't render.
This applies to Checkbox, TextField, TextArea and TextFieldBase.

- `<Checkbox validationState="danger" message="error" …>` → `<Checkbox validationState="danger" validationText="error" …>`
- `<Checkbox message="Check this field" …>` → `<Checkbox helperText="Check this field" …>`
- `<TextField validationState="danger" message="error" …>` → `<TextField validationState="danger" validationText="error" …>`
- `<TextField message="Fill full name" …>` → `<TextField helperText="Fill full name" …>`
- `<TextArea validationState="danger" message="error" …>` → `<TextArea validationState="danger" validationText="error" …>`
- `<TextArea message="Introduce yourself" …>` → `<TextArea helperText="Introduce yourself" …>`
- `<TextFieldBase validationState="danger" message="error" …>` → `<TextFieldBase validationState="danger" validationText="error" …>`
- `<TextFieldBase message="State your address" …>` → `<TextFieldBase helperText="State your address" …>`

Please refer back to these instructions or reach out to our team if you encounter any issues during migration.

### Issue reference

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-desing-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-desing-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig
Comment (literat) [inline packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig:1]: Is the name of the file still valid?

File: packages/web-twig/src/Resources/components/TextArea/README.md [hunk 1]
```diff
@@ -46,27 +46,26 @@ Without lexer:
...
+| `name`              | `string`                                       | `null`  | no       | TextArea name                                                                                               |
+| `placeholder`       | `string`                                       | `null`  | no       | TextArea placeholder                                                                                        |
+| `rows`              | `number`                                       | `null`  | no       | Number of visible rows                                                                                      |
+| `validationState`   | [Validation dictionary][dictionary-validation] | `null`  | no       | Type of validation state.                                                                                   |
+| `validationText`    | `string`, `string[]`                           | `null`  | no       | Validation text                                                                                             |
```
Comment (literat) [inline packages/web-twig/src/Resources/components/TextArea/README.md:67]: Missing `UNSAFE_validationText`?

File: packages/web-twig/src/Resources/components/TextField/README.md [hunk 2]
```diff
@@ -65,17 +65,17 @@ Without lexer:
 | `helperText`        | `string`                                                      | `null`  | no       | Custom helper text                                                      |
```
Comment (literat) [inline packages/web-twig/src/Resources/components/TextField/README.md:65]: `UNSAFE_helperText` missing.

File: packages/web-twig/src/Resources/components/TextField/README.md [hunk 3]
```diff
@@ -65,17 +65,17 @@ Without lexer:
...
 | `name`              | `string`                                                      | `null`  | no       | Input name                                                              |
 | `placeholder`       | `string`                                                      | `null`  | no       | Input placeholder                                                       |
-| `inputWidth`        | `number`                                                      | `null`  | no       | Input width                                                             |
 | `type`              | `email`, `number`, `password`, `search`, `tel`, `text`, `url` | `text`  | no       | Input type                                                              |
 | `validationState`   | [Validation dictionary][dictionary-validation]                | `null`  | no       | Type of validation state.                                               |
+| `validationText`    | `string`, `string[]`                                          | `null`  | no       | Validation text                                                         |
```
Comment (literat) [inline packages/web-twig/src/Resources/components/TextField/README.md:78]: `UNSAFE_validationText` missing.

File: packages/web-twig/src/Resources/components/TextField/stories/TextFieldFluid.twig [hunk 4]
```diff
@@ -2,7 +2,6 @@
     id="textfieldFluid"
     isFluid
     label="Label"
-    message="Message"
```
Comment (literat) [inline packages/web-twig/src/Resources/components/TextField/stories/TextFieldFluid.twig:5]: Replace with `validationText`?

File: packages/web-twig/src/Resources/components/TextFieldBase/README.md [hunk 5]
```diff
@@ -58,30 +58,30 @@ Without lexer:
...
-| `type`              | `email`, `number`, `password`, `search`, `tel`, `text`, `url` | `text`  | no       | Input type                                                              |
-| `validationState`   | [Validation dictionary][dictionary-validation]                | `null`  | no       | Type of validation state.                                               |
-| `value`             | `string`                                                      | `null`  | no       | Input value                                                             |
+| Prop name               | Type                                                          | Default | Required | Description                                                             |
+| ----------------------- | ------------------------------------------------------------- | ------- | -------- | ----------------------------------------------------------------------- |
+| `UNSAFE_helperText`     | `string`                                                      | `null`  | no       | Unescaped custom helper text                                            |
```
Comment (literat) [inline packages/web-twig/src/Resources/components/TextFieldBase/README.md:63]: ABC

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig
Comment (crishpeen) [reply packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig:1]: It is, why wouldn't it be? There is a separate issue to remove this feature class.

File: packages/web-twig/src/Resources/components/TextField/stories/TextFieldFluid.twig (hunk 4 above)
Comment (crishpeen) [reply packages/web-twig/src/Resources/components/TextField/stories/TextFieldFluid.twig:5]: This story is supposed to show the Fluid version. So I tried to showcase only that. If I would add validationText, I would have to add a validationState too. And I don't think it is necessary a if so, I would prefer to have it as a second example in Fluid Section.

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig
Comment (literat) [reply packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig:1]: Ok.

File: packages/web-twig/src/Resources/components/TextArea/README.md (hunk 1 above)
Comment (literat) [reply packages/web-twig/src/Resources/components/TextArea/README.md]: Fixed.

File: packages/web-twig/src/Resources/components/TextField/README.md (hunk 2 above)
Comment (literat) [reply packages/web-twig/src/Resources/components/TextField/README.md]: Fixed.

File: packages/web-twig/src/Resources/components/TextField/README.md (hunk 3 above)
Comment (literat) [reply packages/web-twig/src/Resources/components/TextField/README.md]: Fixed.

File: packages/web-twig/src/Resources/components/TextFieldBase/README.md (hunk 5 above)
Comment (literat) [reply packages/web-twig/src/Resources/components/TextFieldBase/README.md]: Fixed.

File: packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldDisabled.twig [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
+    helperText="Helper text"
+    id="checkboxfieldDisabledHelperText"
     isChecked
     isDisabled
-    label="Checkbox selected"
-    name="checkboxfield7"
+    isRequired
+    label="Checkbox Label"
+    name="checkboxfieldDisabledHelperText"
 />
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldDisabled.twig:16]: Please add a demo of the indeterminate state like we have in the `web` package:

obrazek

File: packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldIndeterminate.twig [hunk 7]
```diff
@@ -1,13 +1,16 @@
...
+<script>
+    const inputs = ['checkboxfieldIndeterminate'];
+
+    window.addEventListener('DOMContentLoaded', function () {
+        inputs.forEach(function (input) {
+            const checkbox = document.getElementById(input);
+            checkbox.indeterminate = true;
+        });
+    });
+</script>
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldIndeterminate.twig:16]: The script may need to be moved once we have indeterminate checkboxes also in other sections…

File: packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldItem.twig [hunk 8]
```diff
@@ -1,13 +1,40 @@
...
+    helperText="Helper text"
+    id="checkboxfieldItemDisabledHelperText"
     isDisabled
     isItem
-    label="Checkbox Item unselected disabled"
-    name="checkboxfield19"
+    isRequired
+    label="Checkbox Label"
+    name="checkboxfieldItemDisabledHelperText"
 />
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldItem.twig:40]: - ⚠️ label text should be bold
- checkbox should be checked

Web:

obrazek

Twig:

obrazek

File: packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldValidation.twig [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    id="checkboxfieldWarning"
+    label="Checkbox Label"
+    validationText="Warning validation text"
+    name="checkboxfieldWarning"
+    validationState="warning"
+/>
+
+<CheckboxField
+    id="checkboxfieldDanger"
+    isChecked
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldValidation.twig:18]: We don't have this one checked in the `web` package…

obrazek

Comment (literat) [reply packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldValidation.twig]: Fixed.

File: packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldItem.twig (hunk 8 above)
Comment (literat) [reply packages/web-twig/src/Resources/components/CheckboxField/stories/CheckboxFieldItem.twig]: Fixed.

File: packages/web-twig/src/Resources/components/Field/README.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
+# ValidationText
+
+The validationText subcomponent displays validation texts for Field components like TextField, TextArea, CheckboxField, FileUploader, etc.
```
[the remark this answers is not in the document]
Comment (adamkudrna) [reply packages/web-twig/src/Resources/components/Field/README.md:3]: OK, it's the same in React.

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaHiddenLabel.twig
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextArea/stories/TextAreaHiddenLabel.twig:1]: The field is filled in the web package:

obrazek

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig [hunk 11]
```diff
@@ -1,22 +1,15 @@
...
+    placeholder="Placeholder"
+/>
+<TextArea
+    id="textareaDisabledFilledV1"
+    isDisabled
+    label="Label"
+    name="textareaDisabledFilledV1"
+    placeholder="Placeholder"
+    value="Filled"
+/>
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextArea/stories/TextAreaV1DisabledFeature.twig:15]: In `web`, this one is `required`:

obrazek

File: packages/web-twig/src/Resources/components/TextArea/stories/TextFieldPasswordToggle.twig
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextArea/stories/TextFieldPasswordToggle.twig:1]: This file is a mistake, isn't it?

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaInline.twig [hunk 12]
```diff
@@ -1,9 +1,9 @@
-<div style="display: flex; gap: 1rem; align-items: flex-end">
+<div style="display: flex; gap: 1rem; align-items: flex-start">
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextArea/stories/TextAreaInline.twig:1]: There is a different alignment in the web package:

obrazek

File: packages/web-twig/src/Resources/components/TextArea/stories/TextAreaValidation.twig [hunk 13]
```diff
@@ -1,38 +1,37 @@
...
     value="Filled"
 />
 <TextArea
-    id="textareaSuccess"
-    isRequired
+    id="textareaDanger"
     label="Label"
-    message="Validation message"
-    name="textareaSuccess"
+    validationText="{{ ['First validation text', 'Second validation text'] }}"
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextArea/stories/TextAreaValidation.twig:21]: The messages are slightly different in the web package:

obrazek

File: packages/web-twig/src/Resources/components/TextField/stories/TextFieldValidation.twig [hunk 14]
```diff
@@ -1,56 +1,37 @@
...
     value="Filled"
 />
 <TextField
-    id="textfieldDisabled"
-    isDisabled
+    id="textfieldDanger"
     label="Label"
-    message="Message"
-    name="textfieldDisabled"
+    validationText="{{ ['First validation text', 'Second validation text'] }}"
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextField/stories/TextFieldValidation.twig:21]: The messages are slightly different in the web package:

obrazek

File: packages/web-twig/src/Resources/components/TextFieldBase/README.md [hunk 15]
```diff
@@ -18,7 +18,7 @@ Default TextFieldBase:
   id="example2"
   isRequired
   label="Label"
-  message="validation failed"
+  validationText="validation failed"
   name="example2"
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextFieldBase/README.md:22]: There are still numbered IDs and names.

File: packages/web-twig/src/Resources/components/TextField/stories/TextFieldV1DisabledFeature.twig [hunk 16]
```diff
@@ -1,28 +1,15 @@
...
-    />
-</div>
+<TextField
+    id="textfieldDisabledV1"
+    isDisabled
+    label="Label"
+    name="textfieldDisabledV1"
+    placeholder="Placeholder"
+/>
+<TextField
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/TextField/stories/TextFieldV1DisabledFeature.twig:8]: This one is marked as required in the web package:

obrazek

**Review by adamkudrna — APPROVED** [review]
👏🏻

**Comment by adamkudrna** [comment]
> And also make it depend on `validationState` and introduce ValidationText component.
> 
> Improve demos or related components.
> 
> Instead of `message` prop use `validationText`. Use `helperText` instead of `validationText` to show permanent helper. If `validationState` is not set `validationText` won't render. This applies to Checkbox, TextField, TextArea and TextFieldBase.
> 
>     * `<Checkbox validationState="danger" message="error" …>` → `<Checkbox validationState="danger" validationText="error" …>`
> 
>     * `<Checkbox message="Check this field" …>` → `<Checkbox helperText="Check this field" …>`
> 
>     * `<TextField validationState="danger" message="error" …>` → `<TextField validationState="danger" validationText="error" …>`
> 
>     * `<TextField message="Fill full name" …>` → `<TextField helperText="Fill full name" …>`
> 
>     * `<TextArea validationState="danger" message="error" …>` → `<TextArea validationState="danger" validationText="error" …>`
> 
>     * `<TextArea message="Introduce yourself" …>` → `<TextArea helperText="Introduce yourself" …>`
> 
>     * `<TextFieldBase validationState="danger" message="error" …>` → `<TextFieldBase validationState="danger" validationText="error" …>`
> 
>     * `<TextFieldBase message="State your address" …>` → `<TextFieldBase helperText="State your address" …>`
> 
> 
> Please refer back to these instructions or reach out to our team if you encounter any issues during migration.

Maybe a few improvements before we merge it:

1. Add structure by using a list for all changes, like:
  - Rename `message` prop to `validationText` in all Form Fields.
  - Introduce ValidationText component.
  - Make `validationText` dependent on `validationState`. _(BTW, what does it mean…?)_
  - Improve demos of related components.
 2. List _all_ affected components in _This applies to …_ I think Select is missing?
3. Fix the migration guide where `message` incorrectly becomes `helperText`.
4. Maybe structure the guide either by change type or by component.
5. Alternatively, simplify the guide so it doesn't mention components at all, for example: `message="Check this field" → `validationText="Check this field"`

**Comment by literat** [comment]
> > And also make it depend on `validationState` and introduce ValidationText component.
> > Improve demos or related components.
> > Instead of `message` prop use `validationText`. Use `helperText` instead of `validationText` to show permanent helper. If `validationState` is not set `validationText` won't render. This applies to Checkbox, TextField, TextArea and TextFieldBase.
> > ```
> > * `<Checkbox validationState="danger" message="error" …>` → `<Checkbox validationState="danger" validationText="error" …>`
> > 
> > * `<Checkbox message="Check this field" …>` → `<Checkbox helperText="Check this field" …>`
> > 
> > * `<TextField validationState="danger" message="error" …>` → `<TextField validationState="danger" validationText="error" …>`
> > 
> > * `<TextField message="Fill full name" …>` → `<TextField helperText="Fill full name" …>`
> > 
> > * `<TextArea validationState="danger" message="error" …>` → `<TextArea validationState="danger" validationText="error" …>`
> > 
> > * `<TextArea message="Introduce yourself" …>` → `<TextArea helperText="Introduce yourself" …>`
> > 
> > * `<TextFieldBase validationState="danger" message="error" …>` → `<TextFieldBase validationState="danger" validationText="error" …>`
> > 
> > * `<TextFieldBase message="State your address" …>` → `<TextFieldBase helperText="State your address" …>`
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please refer back to these instructions or reach out to our team if you encounter any issues during migration.
> 
> Maybe a few improvements before we merge it:
> 
> 1. Add structure by using a list for all changes, like:
> 
> * Rename `message` prop to `validationText` in all Form Fields.
> * Introduce ValidationText component.
> * Make `validationText` dependent on `validationState`. _(BTW, what does it mean…?)_
> * Improve demos of related components.
> 
> 2. List _all_ affected components in _This applies to …_ I think Select is missing?
> 3. Fix the migration guide where `message` incorrectly becomes `helperText`.
> 4. Maybe structure the guide either by change type or by component.
> 5. Alternatively, simplify the guide so it doesn't mention components at all, for example: `message="Check this field"` → `validationText="Check this field"`

I have changed this according to our discussion.

**Comment by adamkudrna** [comment]
The commit message:

```
Feat(web): Unify TextArea demo with Twig
```

should be

```
Docs(web): Unify TextArea demo with Twig
```

**Pull request merged**